gbpservice/nfp/orchestrator/modules/service_orchestrator.py

```diff
@@ -799,45 +815,64 @@ def handle_device_created(self, event):
         return
 
     def handle_device_active(self, event):
-        request_data = event.data
+        nfp_context = event.data
+
+        network_function_instance = nfp_context['network_function_instance']
+        network_function_device = nfp_context['network_function_device']
+        network_function = nfp_context['network_function']
+
+        request_data = {'network_function_device_id': network_function_device['id'],
+                        'network_function_instance_id': network_function_instance['id']}
+
         nfi = {
             'status': nfp_constants.ACTIVE,
             'network_function_device_id': request_data[
                 'network_function_device_id']
         }
         nfi = self.db_handler.update_network_function_instance(
             self.db_session, request_data['network_function_instance_id'], nfi)
-        network_function = self.db_handler.get_network_function(
-            self.db_session, nfi['network_function_id'])
+        network_function_instance['status'] = nfp_constants.ACTIVE
+        network_function_instance['network_function_device_id'] = network_function_device['id']
+
         service_config = network_function['service_config']
+        nfp_core_context.store_nfp_context(nfp_context)
 
         self.create_network_function_user_config(network_function['id'],
                                                  service_config)
 
     def apply_user_config(self, event):
         request_data = event.data
+        nfp_context = event.data['nfp_context']
+
+        network_function = nfp_context['network_function']
         network_function_details = self.get_network_function_details(
-            request_data['network_function_id'])
-        request_data['heat_stack_id'] = self.config_driver.apply_config(
-            network_function_details)  # Heat driver to launch stack
-        network_function = network_function_details['network_function']
+            network_function['id'])
+        request_data['heat_stack_id'], heat_client = self.config_driver.apply_heat_config(
```

Comment:
because dont want to touch apply_config, also helps you to compare what is the change. Lack of better name, suggest if you have any,